### PR TITLE
New tutorial section addressing iszerofunc from matrices.

### PR DESCRIPTION
**Pull request by sylee957**
#### References to other Issues or PRs

#15169

#### Brief description of what is fixed or changed

I have added an additional section `Zero Testing` to address the issues caused by `_iszero`, and also to offer tutorials about troubleshooting the specific case. 
However, I have moved the tutorial under a new section `Possible Issues` instead of `Advanced Methods`, because it had became lengthy after adding all the necessary things to summarize the discoveries.

I have originally created the raw markdown newlined after some chunks separated by comma, to easily trace down changes of words with git,
but it may be changed to square alignment after every grammatical error checked.

I have also added footnotes to reference, as the format I have found from other tutorial sections.

#### Release Notes

- matrices
  - Added a new tutorial section regarding `iszerofunc`.

### CODE REVIEW

File: doc/src/tutorial/matrices.rst [hunk 1]
```diff
@@ -406,3 +406,139 @@ expensive to calculate.
...
+    ...         result = x.is_zero
+    ...     except AttributeError:
+    ...         result = None
+    ...
+    ...     # Warnings if evaluated into None
+    ...     if result == None:
+    ...         warnings.warn("Zero testing of {} evaluated into {}".format(x, result))
+    ...     return result
+    ...
+    >>> m.nullspace(iszerofunc=my_iszero)
```
[the remark this answers is not in the document]
Comment (jksuom) [reply doc/src/tutorial/matrices.rst]: I think that it should be at the end of this line.

[the remark this answers is not in the document]
Comment (jksuom) [reply doc/src/tutorial/matrices.rst]: The documentation is not very clear. In SymPy, there is the [docstring of `sympytestfile`](https://github.com/sympy/sympy/blob/master/sympy/utilities/runtests.py#L923). The flags are explained in [Python docs](https://docs.python.org/2.7/library/doctest.html#option-flags).

**Comment by jksuom** [comment]
This looks good otherwise but many text lines are too long to fit on a 80 character terminal display. (Footnote lines with long urls could probably be left as is.)

File: doc/src/tutorial/matrices.rst [hunk 2]
```diff
@@ -470,10 +470,12 @@ by injecting a custom zero test with warnings enabled.
     __main__:9: UserWarning: Zero testing of -(4*cosh(q/3)**2 - 1)*exp(-q) - 2*cosh(q/3) - exp(-q) evaluated into None
     []
 
-In this case, ``(-exp(q) - 2*cosh(q/3))*(-2*cosh(q/3) - exp(-q)) - (4*cosh(q/3)**2 - 1)**2`` should yield zero,
-but the zero testing had failed to catch.
+In this case,
+ ``(-exp(q) - 2*cosh(q/3))*(-2*cosh(q/3) - exp(-q)) - (4*cosh(q/3)**2 - 1)**2``
```
Comment (jksuom) [inline doc/src/tutorial/matrices.rst]: Should a space be removed from the beginning?

Comment (sylee957) [reply doc/src/tutorial/matrices.rst]: @jksuom 
I have fixed

**Comment by jksuom** [comment]
Thanks, I'll merge in 24 hours.

**Pull request merged**